MERLin/MERLin/Routing/RouterProtocol.swift

```diff
@@ -150,12 +147,24 @@ public extension Router {
         return viewController
     }
     
-    internal func closeButton(for viewController: UIViewController, onClose: (() -> Void)?) -> UIBarButtonItem {
-        let button = UIBarButtonItem(title: closeButtonString, style: .plain, target: nil, action: nil)
+    internal func closeButton(for viewController: UIViewController, closeButtonType: CloseButtonType) -> UIBarButtonItem? {
+        let button: UIBarButtonItem
+        let onCloseAction: (() -> Void)?
```

Comment:
I would put `onCloseAction` to CloseButtonType as computed variable

Reply:
Good idea.